### ISSUE TITLE
Implement RelationalAlgebraFrozenSet difference in pandas

### ISSUE THREAD

**Issue by jonasrenault** [NeuroLang/NeuroLang#607]
Difference of RelationalAlgebraFrozenSet is currently done by relying on python sets' difference, which seems to work fine in most cases, but does not work when using difference on dee sets.

Example
```python
>>> from neurolang.utils.relational_algebra_set.pandas import RelationalAlgebraFrozenSet
>>> dee = RelationalAlgebraFrozenSet.dee()
>>> dee - dee
Empty DataFrame
Columns: []
Index: [0]
>>> len(dee - dee)
1
```

Expected output should be `{}`.

This is causing problems later on in the chase algorithm when sets that should be empty are not.

Difference does work as expected for `NamedRelationalAlgebraFrozenSet`:
```python
>>> from neurolang.utils.relational_algebra_set.pandas import NamedRelationalAlgebraFrozenSet
>>> dee = NamedRelationalAlgebraFrozenSet.dee()
>>> dee - dee
{}
```

**Comment by demianw** [comment]
Good catch! Can you make a PR to solve it please?

**Comment by jonasrenault** [comment]
Yes

**Comment by demianw** [comment]
I think I went too fast in closing this issue. This can bring bad performance problems if the set difference is needed. As the title is very generic, I think it's best to keep this open until we implement the full pandas version of the operation. However merging quickly PR #608 was good to ensure that when we went from using mutable sets to frozen sets, meaning `-=` to `-` the system is still working.
